internal/pkg/agent/application/dispatcher.TestActionDispatcher tests are flaky

Recent examples:

* https://fleet-ci.elastic.co/job/elastic-agent/job/elastic-agent-mbp/job/main/783/

```
=== RUN   TestActionDispatcher/Cancel_queued_action
    dispatcher_test.go:253: PASS:	Handle(string,string,string)
    dispatcher_test.go:254: FAIL:	Save()
        		at: [dispatcher_test.go:232]
    dispatcher_test.go:254: PASS:	DequeueActions()
    dispatcher_test.go:254: FAIL: 1 out of 2 expectation(s) were met.
        	The code you are testing needs to make 1 more call(s).
        	at: [dispatcher_test.go:254]
    --- FAIL: TestActionDispatcher/Cancel_queued_action (0.00s)
```

```
=== RUN   TestActionDispatcher
--- FAIL: TestActionDispatcher (0.54s)
```

--- a/internal/pkg/agent/application/dispatcher/dispatcher_test.go
+++ b/internal/pkg/agent/application/dispatcher/dispatcher_test.go
@@ -9,6 +9,7 @@ import (
 	"testing"
 	"time"
 
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/require"
 
@@ -250,6 +251,7 @@ func TestActionDispatcher(t *testing.T) {
 		case <-time.After(200 * time.Microsecond):
 			// we're not expecting any reset,
 		}
+		assert.Eventuallyf(t, func() bool { return len(def.Calls) > 0 }, 100*time.Millisecond, 100*time.Microsecond, "mock handler for cancel actions has not been called")
 		def.AssertExpectations(t)
 		queue.AssertExpectations(t)
 	})

